src/AstGen.zig

```diff
@@ -1934,11 +1935,12 @@ fn unusedResultExpr(gz: *GenZir, scope: *Scope, statement: Ast.Node.Index) Inner
         // in the above while loop.
         const zir_tags = gz.astgen.instructions.items(.tag);
         switch (zir_tags[inst]) {
-            // For some instructions, swap in a slightly different ZIR tag
+            // For some instructions, modify the zir data
             // so we can avoid a separate ensure_result_used instruction.
-            .call_chkused => unreachable,
             .call => {
-                zir_tags[inst] = .call_chkused;
+                const extra_index = gz.astgen.instructions.items(.data)[inst].pl_node.payload_index;
+                const extra = @ptrCast(*Zir.Inst.Call, &gz.astgen.extra.items[extra_index]);
```

Comment:
@andrewrk I know you aren't a huge fan of using ptrCast on zir, can you recommend a better way to do this?

Reply:
I'm OK with this but `Zir.Inst.Call` needs to become an `extern struct` to make it sound.

Reply:
I ended up just putting the flags first, so they can easily be found without needing to reinterpret anything.